GravatarApp/AvatarPicker/AvatarPickerViewModel.swift

```diff
@@ -231,6 +215,113 @@ class AvatarPickerViewModel: ObservableObject {
             }
         }
     }
+
+    func upload(_ image: UIImage) async {
+        // SwiftUI doesn't update the UI if the grid is empty.
+        // objectWillChange forces the update.
+        objectWillChange.send()
+
+        let localID = UUID().uuidString
+
+        let localImageModel = AvatarImageModel(id: localID, source: .local(image: image), state: .loading, isSelected: false, altText: "")
+        grid.append(localImageModel)
+
+        await doUpload(squareImage: image, localID: localID, accessToken: authToken)
+    }
+
+    func retryUpload(of localID: String) async {
+        guard
+            let model = grid.avatars.first(where: { $0.id == localID }),
+            let localImage = model.localUIImage
+        else {
+            return
+        }
+        grid.setState(to: .loading, onAvatarWithID: localID)
+        await doUpload(squareImage: localImage, localID: localID, accessToken: authToken)
+    }
+
+    func deleteFailed(_ id: String) {
+        withAnimation {
+            _ = grid.deleteModel(id)
+        }
+    }
+
+    private func doUpload(squareImage: UIImage, localID: String, accessToken: String) async {
+        do {
+            guard let avatar = try await avatarService.upload(
+                squareImage,
+                selectionBehavior: .selectUploadedImageIfNoneSelected(for: .hashID(profile.hash)),
+                accessToken: accessToken
+            ) as? AvatarDetails else {
+                fatalError()
+            }
```

Comment:
We need to return `AvatarDetails` from `avatarService.upload()`
This is a breaking change on the SKD

Reply:
> all better than having a fatalError() call in production.

Yes I didn't mean to ship this fatal error to production, but I used it while we decide how to approach the SDK change. 
As mentioned in the description:
>On this PR (so far) we are force-casting it, which is guaranteed to succeed, but ideally we need to get a AvatarDetails type from the upload.